query-engine/core/src/query_graph_builder/extractors/filters/scalar.rs

```diff
@@ -440,7 +440,7 @@ impl<'a> ScalarFilterParser<'a> {
                         Ok(ConditionValue::reference(field_ref))
                     }
                     Some(Field::Scalar(field_ref)) => Err(QueryGraphBuilderError::InputError(format!(
-                        "Expected a referenced scalar field of type {}{} but found {} of type {}{}.",
+                        "Expected a referenced scalar field of type {:?}{} but found {} of type {:?}{}.",
```

Comment:
We could pass down some context and use the display impl here too, if that sounds like a better solution.